src/modules/book/book.service.ts

```diff
@@ -0,0 +1,125 @@
+import {
+    Injectable,
+    InternalServerErrorException,
+    NotFoundException,
+    BadRequestException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { In, Repository } from 'typeorm';
+import { Book } from './entities/book.entity';
+import { CreateBookDto } from './dto/create-book.dto';
+import { Author } from '../author/entities/author.entity';
+import { Genre } from '../genre/entities/genre.entity';
+import { UpdateBookDto } from './dto/update-book.dto';
+
+@Injectable()
+export class BookService {
+    constructor(@InjectRepository(Book) private bookRepo: Repository<Book>) {}
+
+    async findAll(): Promise<Book[]> {
+        try {
+        return await this.bookRepo.find({
+            relations: ['author', 'bookInstances', 'genres'],
+        });
+        } catch (error) {
+        throw new InternalServerErrorException(
+            `Failed to fetch books: ${error.message}`,
+        );
+        }
+    }
+
+    async findOneById(id: number): Promise<Book> {
+        const book = await this.bookRepo.findOne({
+        where: { id },
+        relations: ['author', 'genres', 'bookInstances'],
+        });

```

Comment:
convention -> kiểm tra lại indent trong các block statement e nha
```suggestion
        const book = await this.bookRepo.findOne({
            where: { id },
            relations: ['author', 'genres', 'bookInstances'],
        });

```